src/kernels/tensor_view.hpp

```diff
@@ -77,4 +77,20 @@ struct tensor_layout_t
     uint64_t layout[N];
 };
 
+template <typename T>
+__host__ __device__ T inline get2DVal(const T* x, tensor_view_t<2> x_tv, uint64_t idx)
+{
+    tensor_layout_t<2> layout = tensor_layout_t<2>(x_tv, idx);
+    uint64_t x_idx            = x_tv.get_tensor_view_idx(layout);
+    return x[x_idx];
+}
+
+template <typename T>
+__host__ __device__ void inline set1DVal(T* x, tensor_view_t<1> x_tv, uint64_t idx, T val)
```

Comment:
Please add N to the template to create a more general setval function.

Reply:
I have updated template to receive N